### PR TITLE
Dblatcher/render requests

### PR DESCRIPTION
**Pull request by dblatcher**
## What does this change?

 - Adds an environment variable to allow using a local version of email-rendering instead of PROD
 - new API routes to fetch demo content from email-rendering by posting `newsletterData` to it
- adds preview pages to show the demo content for a launched newsletter
- puts the preview alongside the 'edit rendering option' form so the user can see the effect of the changes

## How to test

try the new pages.

## How can we measure success?

Uses better able to understand how the changes will effect the design.

## Have we considered potential risks?

Should do some user acceptance testing on CODE before merging? 

## Images

Screenshot 2023-07-19 at 12 54 46

### CODE REVIEW

File: apps/newsletters-api/src/app/routes/rendering-templates.ts [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
 export type RenderingTemplate = {
 	id: string;
 	status: string;
 	title: string;
 };
 
-const TEMPLATES_LIST_URL =
-	'https://email-rendering.guardianapis.com/info/templates/';
+const NEWSLETTER_RENDER_URL = `${getEmailRenderingHost()}/data-article/render-template`;
+const TEMPLATES_LIST_URL = `${getEmailRenderingHost()}/info/templates/`;
```
Comment (phillipbarron) [inline apps/newsletters-api/src/app/routes/rendering-templates.ts:19]: We could avoid evaluating that twice with an assign

```suggestion
const EMAIL_RENDERING_HOST = getEmailRenderingHost();
const NEWSLETTER_RENDER_URL = `${EMAIL_RENDERING_HOST}/data-article/render-template`;
const TEMPLATES_LIST_URL = `${EMAIL_RENDERING_HOST}/info/templates/`;

```

File: apps/newsletters-ui/src/app/components/TemplatePreview.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import RefreshIcon from '@mui/icons-material/Refresh';
```
Comment (phillipbarron) [inline apps/newsletters-ui/src/app/components/TemplatePreview.tsx:1]: what github says :)

**Review by phillipbarron — APPROVED** [review]
This is really great, awesome job.

Tiny nit is that would could adfd a few pixels of padding below above the preview as it's right up against the series tag notification which looks a little busy to my eye.

love it!

**Comment by dblatcher** [comment]
> Tiny nit is that would could adfd a few pixels of padding below above the preview as it's right up against the series tag notification which looks a little busy to my eye.

👍 I'm also thinking we should let this page be wider than the others - the `ContentWrapper` limits the page width to 'lg', but I could add a width prop so this page can be 'xl'.

**Comment by dblatcher** [comment]
> > > Tiny nit is that would could adfd a few pixels of padding below above the preview as it's right up against the series tag notification which looks a little busy to my eye.
> > 
> > 
> > 👍 I'm also thinking we should let this page be wider than the others - the `ContentWrapper` limits the page width to 'lg', but I could add a width prop so this page can be 'xl'.
> 
> Yeah, it is a little squashed - give it a try 👍

Pushed some layout changes - I also used a MUI elevated Card instead of having the outline on the preview container

Screenshot 2023-07-20 at 10 40 14

**Pull request merged**